app/shop.py

```diff
@@ -0,0 +1,33 @@
+from dataclasses import dataclass
+from datetime import datetime
+
+
+@dataclass
+class Shop:
+    name: str
+    location: list[int]
+    products: dict
+
+    def calculate_product_price(self, customer_products: dict) -> float:
+        return sum([
+            value * self.products.get(key)
+            for key, value in customer_products.items()
+        ])
```

Comment:
no need to create list 
```suggestion
        return sum(
            value * self.products.get(key)
            for key, value in customer_products.items()
        )
```